### ISSUE TITLE
Cookie causes Plugin to freeze Chrome tab

### ISSUE THREAD

**Issue by zone5Per** [duckduckgo/duckduckgo-privacy-extension#680]
## Description
First of all, I have already lodged a report using the 'Report Broken Site' link on the extension popup. But on further investigation on my end I think this is actually an issue with the Chrome plugin rather than just a broken site.

When accessing one of our web applications the DuckDuckGo plugin causes the tab to become unresponsive. Only a restart of Chrome or manually killing the Google Chrome Helper (Renderer) task in Activity Monitor that is consuming a very high amount of CPU solves it. 

The issue does not appear when turning off the plugin, site protection or running the plugin in Incognito mode. 

## Steps to Reproduce
1. Go to https://portal.zone5cloud.com/en/sign-up and sign up for a new account, ensure you tick the **"remember me"** checkbox
2. Once logged in either click on your name in the top right corner of screen and select account or manually navigate too https://portal.zone5cloud.com/en/manager/account
3. Page freezes, either restart Chrome or kill Google Chrome Helper (Renderer) in activity monitor to regain control

**Expected behavior:**
User should be navigated to user account page.

**Actual behavior:**
The tab freezes, only by killing the process can you regain control again.

**Versions**
- Extension: 2021.5.10
- Browser: Chrome Version 90.0.4430.212 (Official Build) (x86_64)
- OS: macOS Catalina 10.15.6 (19G2021)

## Additional Information
1. Does not happen in Incognito mode
2. Does not happen when turning off site protection
3. On login we store an email in either session/Cookie storage depending on if user ticked the remember me checkbox or not, but it only breaks when using Cookie storage. By not setting this email cookie it starts working as expected again. The only difference between the session/Cookie storage code is that I use a service to set the Cookie which automatically URI encodes the email when setting it while it is stored without encoding in session storage. By manually setting the Cookie value without the library so it is stored without being URI encoded it seems to work as expected. Like so:

sessionStorage - per@email.com <-- does not cause issue
cookie - per%40email.com <-- DuckDuckGo freezes page
cookie - per@email.com <-- Page loads as expected, this scenario not testable on the website because requires a code change.

For some reason it does not seem to like that encoding of the email in the Cookie value. Also it only seems to be an issue with this particular cookie and not the rest, even though we are setting them in the same way.

I did lodge a broken site issue via the Plugin but this issue with the Cookie and the encoding makes me think this is probably more of a plugin issue. If I'm wrong and this is expected then maybe someone can point me towards some documentation or details around it before closing.

**Comment by sammacbeth** [comment]
Thanks for the report!

I was able to reproduce the issue in chrome, and tracked the issue down to our first-party cookie protection code. The issue is that in this code we resolve a Promise, but your page has injected a polyfilled implementation of `Promise`. This then causes a infinite loop, where the page continuously sets the `z5TokenStore.email` cookie in an infinite loop, crashing the tab.

I've opened #682 which fixes this issue by ensuring that we always use the native `Promise.prototype.then` implementation.
